src/flint/fq_default.jl

```diff
@@ -675,20 +645,50 @@ function neg!(z::FqFieldElem, a::FqFieldElem)
   return z
 end
 
-function mul!(z::FqFieldElem, x::FqFieldElem, y::FqFieldElem)
-  ccall((:fq_default_mul, libflint), Nothing,
-        (Ref{FqFieldElem}, Ref{FqFieldElem}, Ref{FqFieldElem}, Ref{FqField}), z, x, y, y.parent)
+#
+
+function add!(z::FqFieldElem, x::FqFieldElem, y::FqFieldElem)
+  @ccall libflint.fq_default_add(z::Ref{FqFieldElem}, x::Ref{FqFieldElem}, y::Ref{FqFieldElem}, x.parent::Ref{FqField})::Nothing
   z.poly = nothing
   return z
 end
 
-function add!(z::FqFieldElem, x::FqFieldElem, y::FqFieldElem)
-  ccall((:fq_default_add, libflint), Nothing,
-        (Ref{FqFieldElem}, Ref{FqFieldElem}, Ref{FqFieldElem}, Ref{FqField}), z, x, y, x.parent)
+#
+
+function sub!(z::FqFieldElem, x::FqFieldElem, y::FqFieldElem)
+  @ccall libflint.fq_default_sub(z::Ref{FqFieldElem}, x::Ref{FqFieldElem}, y::Ref{FqFieldElem}, x.parent::Ref{FqField})::Nothing
+  z.poly = nothing
+  return z
+end
+
+#
+
+function mul!(z::FqFieldElem, x::FqFieldElem, y::FqFieldElem)
+  @ccall libflint.fq_default_mul(z::Ref{FqFieldElem}, x::Ref{FqFieldElem}, y::Ref{FqFieldElem}, y.parent::Ref{FqField})::Nothing
   z.poly = nothing
   return z
 end
 
+function mul!(z::FqFieldElem, x::FqFieldElem, y::ZZRingElemOrPtr)
+  @ccall libflint.fq_default_mul_fmpz(z::Ref{FqFieldElem}, x::Ref{FqFieldElem}, y::Ref{ZZRingElem}, x.parent::Ref{FqField})::Nothing
+  return z
+end
+
+function mul!(z::FqFieldElem, x::FqFieldElem, y::Int)
+  @ccall libflint.fq_default_mul_si(z::Ref{FqFieldElem}, x::Ref{FqFieldElem}, y::Int, x.parent::Ref{FqField})::Nothing
+  return z
+end
+
+function mul!(z::FqFieldElem, x::FqFieldElem, y::UInt)
+  @ccall libflint.fq_default_mul_ui(z::Ref{FqFieldElem}, x::Ref{FqFieldElem}, y::Int, x.parent::Ref{FqField})::Nothing
+  return z
+end
```

Comment:
```suggestion
function mul!(z::FqFieldElem, x::FqFieldElem, y::ZZRingElemOrPtr)
  @ccall libflint.fq_default_mul_fmpz(z::Ref{FqFieldElem}, x::Ref{FqFieldElem}, y::Ref{ZZRingElem}, x.parent::Ref{FqField})::Nothing
  z.poly = nothing
  return z
end

function mul!(z::FqFieldElem, x::FqFieldElem, y::Int)
  @ccall libflint.fq_default_mul_si(z::Ref{FqFieldElem}, x::Ref{FqFieldElem}, y::Int, x.parent::Ref{FqField})::Nothing
  z.poly = nothing
  return z
end

function mul!(z::FqFieldElem, x::FqFieldElem, y::UInt)
  @ccall libflint.fq_default_mul_ui(z::Ref{FqFieldElem}, x::Ref{FqFieldElem}, y::Int, x.parent::Ref{FqField})::Nothing
  z.poly = nothing
  return z
end
```
I just found `_as_poly` as a place where this `.poly` is ever read from. I am not sure about the intend of this, but I think we should nonetheless set it always to `nothing` whenever we do something mutating.